src/universal/client.tsx

```diff
@@ -44,6 +44,16 @@ if (__DEVTOOLS__) {
 
 ReactDOM.render(
   <Provider store={store} key="provider">
-    {component}
+    <IntlProvider locale={language} messages={messages}>
+      {component}
+    </IntlProvider>
   </Provider>,
   destinationElement);
+
+  if (process.env.NODE_ENV !== 'production') {
+  (window as any).React = React; // enable debugger
+
+  if (!destinationElement || !destinationElement.firstChild || !destinationElement.firstChild.attributes || !destinationElement.firstChild.attributes['data-react-checksum']) {
+    console.error('Server-side React render was discarded. Make sure that your initial render does not contain any client-side code.');
+  }
+}
```

Comment:
This, theoretically, will detect and show a warning if React has to re-render the client again. 

From what I've seen it is very difficult to have a server-side rendered page that React will **not** re-render on the client when you use `material-ui` component framework, because `material-ui` generates styles based on user-agent, and will override some styles almost all the time, therefore react will not have the same `checksum` because the `style` attributes differ.